pkg/install/install.go

```diff
@@ -397,9 +397,31 @@ func (i *Installer) deployARMTemplate(ctx context.Context, rg string, tName stri
 		return err == nil, err
 	}, timeoutCtx.Done())
 
-	if isQuota, errMsg := hasResourceQuotaExceededError(err); isQuota {
-		err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeQuotaExceeded, errMsg, "")
+	if err != nil {
+		marshaled, marshalErr := json.Marshal(err)
+		if marshalErr != nil {
+			return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
+		}
+		errDetails := string(marshaled)
+
+		// Try to give some kind of information even if marshaled error doesn't have any
+		if errDetails == "{}" {
```

Comment:
I'm guessing we'd still want to know about `wait.ErrWaitTimeout`, even though it doesn't have any marshalable fields, for instance.